File: July-LeetCoding-Challenge/12-Reverse-Bits/Reverse-Bits.cpp
class Solution1 {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t r = 0;
        
        for(int i=0; i<32; ++i){
            if(n&(1<<i))
                r = r|(1<<(32-i-1));
        }
        
        return r;
    }
};

File: July-LeetCoding-Challenge/12-Reverse-Bits/Reverse-Bits.kt
package string_integer

class ReverseBitsKotlin190 {
    // you need treat n as an unsigned value
    fun reverseBits(n: Int): Int {
        var current = n
        var result = 0
        for (time in 0 until 32) {
            result = if (current.and(1) == 1) {
                result.shl(1) + 1
            } else {
                result.shl(1)
            }
            current = current.shr(1)
        }
        return result
    }
}